controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -1611,6 +1639,12 @@ private boolean checkCoverage(double keyStart, double keyEnd, TreeMap<Double, Do
                 .thenApply(x -> !x);
     }
 
+    private CompletableFuture<Boolean> isScopeDeleted(String scope, OperationContext context) {
+        return streamMetadataStore.checkScopeInDeletingTable(scope, context, executor)
```

Comment:
This check needs to be performed before all the other operations like SEALING a Stream, Deleting a Stream, creating transactions etc right? This would ensure the controller responds with a correct response if scope deletion is in progress.

Reply:
Not really. I think we should be good if we provide the check only during creation. Because SEAL api call will not have any effect and in case of delete, the event will fail eventually. But we have to mainly restrict the creation.